docs/testing-styleguide.MD

```diff
@@ -0,0 +1,280 @@
+│ │       └── common
+│ │   └── visual
+│ ├── locators
+│ │   └── [component-name]
+│ │       ├── index.js (exported arrow functions for locators)
+│ │       └── locators.js (string const locators)
+│ ├── plugins
+│ └── support
+│     └── step-definitions
+│         └── [component-name]-steps.js (files with cucumber steps)
+├── .eslintrc.json
+├── cypress.env.json
+├── cypress.json
+├── README.md
+└── tsconfig.json
+```
+
+##### Coding Standards
+Cypress tests are written in Gherkin syntax with Cucumber expressions. Observe the following additional standards:
+1.	Use ESlint plugin in Visual Studio Code to make sure code format is preserved.
+2.	Use kebab-case for all `*.js` file names.
+3.	Use camelCase for `*.feature` file names.
+4.	Use Gherkin syntax keywords in feature files:
+  * GIVEN - the Given part describes the state of the world before you begin the behaviour you are specifying in this scenario. You can think of it as the pre-conditions to the test;
+  * WHEN - the When section is the behaviour that you are specifying which is under test. All the actions you need to perform before making an assertion;
+  * THEN - the Then section describes the changes you expect due to the specified behaviour. Regular assertion;
+  * AND - use this to avoid duplication of the keywords (Given/When/Then). Each of the sections could have their own AND state. Indent AND lines under the previous Given/When/Then keyword by two spaces;
+  * Scenario outline - tests that are using parameters from given `Examples` under the test and will run as many times as there are examples;
+  * Scenario - tests that are run once.
+5.	Write step definitions using Cucumber expressions to link to the Gherkin feature files:
+  * Step definitions are created per component as `component-steps.js` files and stored in `carbon/cypress/support/step-definitions`. They should follow the `component.feature` files exactly;
+  * Common steps (`common-steps.js`) are used for all common actions.
+
+Some common test steps:
+  * I set `<knobs field>` to `<parameter>`;
+  * I select `<knobs field>` to `<parameter>`;
+  * I check/uncheck `<knobs field>` checkbox;
+  * I open `<name>` component page;
+    * To open different stories on Storybook;
+      * I open `<name>` component page `default`/`basic`/`with button`/`legacy spinner`/`iframe`/`with button page in iframe`/`multiple`/`validations`.
+
+##### Scenario tags
+Test scenarios in feature files can be tagged to enable a subset of scenarios to be run, ignored or identified in some manner. Use the following tags:
+
+1.	`@positive` for a happy path.
+2.	`@negative` for the negative scenario.
+3.	`@ignore` for temporarily ignored scenarios.
+4.	`@[bug-number]` for example `@FE-1234` - use this tag after @ignore to explain why the test is ignored.
+5.	`@ignore` and `@[bug-number]` should be removed after the bug fix.
+6.	`@accessibility` tests verify accessibility violations.
+7.	`@validations` tests verify validation components.
```

Comment:
also `@validations` needs to be removed from here as I mentioned above.

Reply:
Again, I took this straight from the Cypress README.md and assumed it was current information.

Reply:
I'll update this document once the changes are in `master`.